### PR TITLE
Make solidus_core depend on actionmailer and activerecord instead of rails

### PR DESCRIPTION
**Pull request by BenMorganIO**
We originally closed #2239 because we thought we were going to get it in from #2236 but never did. So... Here's a new PR!

---

Before:

```shell
$ bundle
Bundle complete! 30 Gemfile dependencies, 126 gems now installed.
```

After:

```shell
$ bundle
Bundle complete! 30 Gemfile dependencies, 123 gems now installed.
```

### CODE REVIEW

**Comment by cbrunsdon** [comment]
At the very least I think we'll need to add activesupport, activejob, and railties, which are the three I know we depend on inside of core.

**Comment by cbrunsdon** [comment]
@BenMorganIO aye, `Rails` is from railties, but we don't have that dependency, we get it through other gems that do explicitely have it listed, which I don't think is good to rely on.

**Comment by jhawthorn** [comment]
> At the very least I think we'll need to add activesupport, activejob, and railties, which are the three I know we depend on inside of core.

We do get these from other gems, but it would make sense to add them to our deps

File: core/solidus_core.gemspec [hunk 1]
```diff
@@ -19,7 +19,10 @@ Gem::Specification.new do |s|
   s.required_ruby_version     = '>= 2.2.2'
   s.required_rubygems_version = '>= 1.8.23'
 
-  %w[activerecord actionmailer].each do |rails_dep|
+  %w[
+    actionmailer actionpack actionview activejob activemodel activerecord
+    activesupport
```
Comment (cbrunsdon) [inline core/solidus_core.gemspec]: railties too please

**Pull request merged**